modules/karpenter/main.tf

```diff
@@ -37,7 +37,7 @@ resource "aws_sqs_queue_policy" "this" {
 resource "aws_cloudwatch_event_rule" "this" {
   for_each = { for k, v in local.event_rules : k => v if var.enable_spot_termination }
 
-  name          = each.value.name
+  name_prefix   = "${var.rule_name_prefix}-${each.value.name}-"
```

Comment:
This may cause issues if length of the characters are above the limit (35 characters?). It's better to use try block by making `rule_name_prefix` as null

Reply:
When I look up It is limited to 64 characters.

<img width="774" alt="Screenshot 2023-03-16 at 09 52 35" src="https://user-images.githubusercontent.com/16239096/225501113-ab3e8480-25bc-47dd-9769-584bbe3bf3d2.png">

The longest default-generated name has 46 characters. So I decide to add input validation instead of changing to null but I make the default prefix to empty string. Maybe some users want informative names that look like which rule name is related to which cluster if they have more than one cluster.

<img width="1079" alt="Screenshot 2023-03-16 at 09 26 00" src="https://user-images.githubusercontent.com/16239096/225504426-436fd5c6-8f63-423b-bcf7-dbd728cb0d89.png">

according to the above screenshot, I decided to reserve 50 characters for the default generated name. And 14 characters for the user prefix name. is that ok?

The following screenshot is default name prefix - 

<img width="1561" alt="Screenshot 2023-03-16 at 09 50 46" src="https://user-images.githubusercontent.com/16239096/225505219-bec2b0ab-8778-4e5b-b958-a027c5c2fd59.png">

The following screenshot is with `karpenter_event_rule_name_prefix           = "clusterid-"` -

<img width="1559" alt="Screenshot 2023-03-16 at 09 40 29" src="https://user-images.githubusercontent.com/16239096/225505464-333a1bf2-4e5a-4cac-9363-281552f2e715.png">

This is validation test with more than 14 characters - 

<img width="682" alt="Screenshot 2023-03-16 at 10 00 34" src="https://user-images.githubusercontent.com/16239096/225505575-cd332e8a-e742-467b-8f27-4323c6ce18e4.png">